storage/engine/mvcc.go

```diff
@@ -872,31 +906,48 @@ func MVCCScan(engine Engine, key, endKey proto.Key, max int64, timestamp proto.T
 	if len(endKey) == 0 {
 		return nil, emptyKeyError()
 	}
-	encKey := MVCCEncodeKey(key)
-	encEndKey := MVCCEncodeKey(endKey)
+
+	buf := getBufferPool.Get().(*getBuffer)
+	defer getBufferPool.Put(buf)
+
+	encEndKey := mvccEncodeKey(buf.key[0:0], endKey)
+	keyBuf := buf.key[len(encEndKey):len(encEndKey)]
```

Comment:
It looks like this will result in an index error if endKey is longer than 1024 bytes.